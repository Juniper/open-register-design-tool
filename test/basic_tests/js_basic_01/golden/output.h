//   Ordt 230719.01 autogenerated file 
//   Input: ./js_basic_01/test.js
//   Parms: ./js_basic_01/test.parms
//   Date: Thu Jul 20 13:46:38 EDT 2023
//

#ifndef __SIMPLE1_REGISTER_MAP__
#define __SIMPLE1_REGISTER_MAP__

#include <bits.h>

/* SIMPLE1_REGISTERS memory map */
enum SIMPLE1_REGS {
  STATS_DISABLE_CHECK_0 = 0x1000,
  STATS_DISABLE_CHECK_1 = 0x1004,
  STATS_FEATURES = 0x1100,
  STATS_LOG_ADDRESS1 = 0x1104,
  STATS_LOG_ADDRESS2 = 0x1110,
  STATS_WIDE_REG = 0x1120,
  RDR_REORDER_WINDOW = 0x4000,
  RDR_STREAM_RECONF = 0x4004,
  RDR_CP_FP_WR_BUFFER_0 = 0x61c0,
  RDR_CP_FP_WR_BUFFER_1 = 0x61c4,
  RDR_CP_FP_WR_BUFFER_2 = 0x61c8,
  RDR_CP_FP_WR_BUFFER_3 = 0x61cc,
  RDR_ROLL32_COUNTER_REG = 0x61d0,
  RDR_SAT32_COUNTER_REG = 0x61d4,
  RDR_RCNT_SAT_LOG = 0x61d8,
  EXTRA_REG = 0x61dc
};

/* STATS_DISABLE_CHECK_0 (disable_check register) register fields */
#define STATS_DISABLE_CHECK_0_L0_B3 BIT(6)    /* l0_b3 field */
#define STATS_DISABLE_CHECK_0_L0_B2 BIT(5)    /* l0_b2 field */
#define STATS_DISABLE_CHECK_0_L0_B1 BIT(4)    /* l0_b1 field */
#define STATS_DISABLE_CHECK_0_L0_B0 BIT(3)    /* l0_b0 field */
#define STATS_DISABLE_CHECK_0_LUT_ST BIT(2)    /* lut_st field */
#define STATS_DISABLE_CHECK_0_CBUF_FIFO BIT(1)    /* cbuf_fifo field */
#define STATS_DISABLE_CHECK_0_TIMER_SRAM BIT(0)    /* timer_sram field */

/* STATS_DISABLE_CHECK_1 (disable_check register) register fields */
#define STATS_DISABLE_CHECK_1_L0_B3 BIT(6)    /* l0_b3 field */
#define STATS_DISABLE_CHECK_1_L0_B2 BIT(5)    /* l0_b2 field */
#define STATS_DISABLE_CHECK_1_L0_B1 BIT(4)    /* l0_b1 field */
#define STATS_DISABLE_CHECK_1_L0_B0 BIT(3)    /* l0_b0 field */
#define STATS_DISABLE_CHECK_1_LUT_ST BIT(2)    /* lut_st field */
#define STATS_DISABLE_CHECK_1_CBUF_FIFO BIT(1)    /* cbuf_fifo field */
#define STATS_DISABLE_CHECK_1_TIMER_SRAM BIT(0)    /* timer_sram field */

/* STATS_FEATURES (features register) register fields */
#define STATS_FEATURES_VERSION GENMASK(31, 30)    /* version field */
#define STATS_FEATURES_SYN_LOGGED BIT(9)    /* syn_logged field */
#define STATS_FEATURES_WDTH GENMASK(8, 4)    /* wdth field */
#define STATS_FEATURES_GENERATOR BIT(3)    /* generator field */
#define STATS_FEATURES_CHECKER BIT(2)    /* checker field */
#define STATS_FEATURES_CORRECTABLE BIT(1)    /* correctable field */
#define STATS_FEATURES_ADR_LOGGED BIT(0)    /* adr_logged field */

/* STATS_LOG_ADDRESS1 (log_address1 register) register fields */
#define STATS_LOG_ADDRESS1_VALUE GENMASK(9, 0)    /* value field */

/* STATS_LOG_ADDRESS2 (log_address2 register) register fields */
#define STATS_LOG_ADDRESS2_VALUE GENMASK(9, 0)    /* value field */

/* STATS_WIDE_REG (wide_reg register) register fields */
#define STATS_WIDE_REG_VALUE GENMASK(95, 0)    /* value field */

/* RDR_REORDER_WINDOW (reorder_window register) register fields */
#define RDR_REORDER_WINDOW_WS_ENABLE BIT(31)    /* ws_enable field */
#define RDR_REORDER_WINDOW_SET_BACK GENMASK(26, 16)    /* set_back field */
#define RDR_REORDER_WINDOW_SIZE GENMASK(10, 0)    /* size field */

/* RDR_STREAM_RECONF (stream_reconf register) register fields */
#define RDR_STREAM_RECONF_BUSY BIT(16)    /* busy field */
#define RDR_STREAM_RECONF_STREAM GENMASK(9, 0)    /* stream field */

/* RDR_CP_FP_WR_BUFFER_0 (buffer register) register fields */
#define RDR_CP_FP_WR_BUFFER_0_ADDR0 GENMASK(26, 16)    /* addr0 field */
#define RDR_CP_FP_WR_BUFFER_0_ADDR1 GENMASK(10, 0)    /* addr1 field */

/* RDR_CP_FP_WR_BUFFER_1 (buffer register) register fields */
#define RDR_CP_FP_WR_BUFFER_1_ADDR0 GENMASK(26, 16)    /* addr0 field */
#define RDR_CP_FP_WR_BUFFER_1_ADDR1 GENMASK(10, 0)    /* addr1 field */

/* RDR_CP_FP_WR_BUFFER_2 (buffer register) register fields */
#define RDR_CP_FP_WR_BUFFER_2_ADDR0 GENMASK(26, 16)    /* addr0 field */
#define RDR_CP_FP_WR_BUFFER_2_ADDR1 GENMASK(10, 0)    /* addr1 field */

/* RDR_CP_FP_WR_BUFFER_3 (buffer register) register fields */
#define RDR_CP_FP_WR_BUFFER_3_ADDR0 GENMASK(26, 16)    /* addr0 field */
#define RDR_CP_FP_WR_BUFFER_3_ADDR1 GENMASK(10, 0)    /* addr1 field */

/* RDR_ROLL32_COUNTER_REG (roll32_counter_reg register) register fields */
#define RDR_ROLL32_COUNTER_REG_COUNT GENMASK(3, 0)    /* count field */

/* RDR_SAT32_COUNTER_REG (sat32_counter_reg register) register fields */
#define RDR_SAT32_COUNTER_REG_COUNT GENMASK(3, 0)    /* count field */

/* RDR_RCNT_SAT_LOG (rcnt_sat_log register) register fields */
#define RDR_RCNT_SAT_LOG_EN BIT(31)    /* en field */
#define RDR_RCNT_SAT_LOG_SUBCH GENMASK(30, 26)    /* subch field */
#define RDR_RCNT_SAT_LOG_STR GENMASK(25, 16)    /* str field */
#define RDR_RCNT_SAT_LOG_COUNT GENMASK(15, 0)    /* count field */

/* EXTRA_REG (extra_reg name) register fields */
#define EXTRA_REG_VALUE GENMASK(31, 0)    /* value field */

#endif
