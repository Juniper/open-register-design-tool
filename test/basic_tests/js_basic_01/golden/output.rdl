/*   Ordt 190725.01 autogenerated file 
 *   Input: ./js_basic_01/test.js
 *   Parms: ./js_basic_01/test.parms
 *   Date: Sun Nov 17 17:00:55 EST 2019
 */

// address map simple1
addrmap {

  name = "Registers for simple1";
  // register set stats
  regfile {
  
    name = "stats register_set";
    // register disable_check
    reg {
    
      name = "disable_check register";
      field {
        name = "timer_sram field";
        sw=rw; 
      } timer_sram [0:0] =0x0;
      
      field {
        name = "cbuf_fifo field";
        sw=rw; 
      } cbuf_fifo [1:1] =0x0;
      
      field {
        name = "lut_st field";
        sw=rw; 
      } lut_st [2:2] =0x0;
      
      field {
        name = "l0_b0 field";
        sw=rw; 
      } l0_b0 [3:3] =0x0;
      
      field {
        name = "l0_b1 field";
        sw=rw; 
      } l0_b1 [4:4] =0x0;
      
      field {
        name = "l0_b2 field";
        sw=rw; 
      } l0_b2 [5:5] =0x0;
      
      field {
        name = "l0_b3 field";
        sw=rw; 
      } l0_b3 [6:6] =0x0;
      
    } disable_check[2] @0x0;
    
    // register features
    reg {
    
      name = "features register";
      field {
        name = "adr_logged field";
        sw=r; 
      } adr_logged [0:0] =0x1;
      
      field {
        name = "correctable field";
        sw=r; 
      } correctable [1:1] =0x0;
      
      field {
        name = "checker field";
        sw=r; 
      } checker [2:2] =0x1;
      
      field {
        name = "generator field";
        sw=r; 
      } generator [3:3] =0x1;
      
      field {
        name = "wdth field";
        sw=r; 
      } wdth [8:4] =0x7;
      
      field {
        name = "syn_logged field";
        sw=r; 
      } syn_logged [9:9] =0x0;
      
      field {
        name = "version field";
        sw=r; 
      } version [31:30] =0x2;
      
    } features @0x100;
    
    // register log_address1
    reg {
    
      name = "log_address1 register";
      desc = "
        newdesc bla bla
      ";
      category = "STATE";
      field {
        name = "value field";
        sw=rw; 
      } value [9:0] =0x0;
      
    } log_address1 @0x104;
    
    // register log_address2
    reg {
    
      name = "log_address2 register";
      desc = "
        newdesc 2 bla bla
      ";
      category = "STATE";
      field {
        name = "value field";
        sw=rw; 
      } value [9:0] =0x0;
      
    } log_address2 @0x110;
    
    // register wide_reg
    reg {
    
      name = "wide_reg register";
      regwidth = 128;
      field {
        name = "value field";
        sw=rw; 
      } value [95:0] =0xa5a5a5a5a5a5a5a5a5a5a5a5;
      
    } wide_reg @0x120;
    
  } stats @0x1000;
  
  // register set rdr
  regfile {
  
    name = "rdr register_set";
    // register reorder_window
    reg {
    
      name = "reorder_window register";
      category = "STATE";
      field {
        name = "size field";
        desc = "
          SW Configuration field
        ";
        sw=rw; 
      } size [10:0] =0x80;
      
      field {
        name = "set_back field";
        desc = "
          SW Configuration field
        ";
        sw=rw; 
      } set_back [26:16] =0x10;
      
      field {
        name = "ws_enable field";
        desc = "
          SW Configuration field
        ";
        sw=rw; 
      } ws_enable [31:31] =0x1;
      
    } reorder_window @0x0;
    
    // register stream_reconf
    reg {
    
      name = "stream_reconf register";
      field {
        name = "stream field";
        desc = "
          SW Configuration field
        ";
        sw=rw; 
      } stream [9:0] =0x0;
      
      field {
        name = "busy field";
        desc = "
          HW Status field
        ";
        sw=r; 
      } busy [16:16] =0x0;
      
    } stream_reconf @0x4;
    
    // register set cp_fp_wr
    regfile {
    
      name = "cp_fp_wr register_set";
      // register buffer
      reg {
      
        name = "buffer register";
        category = "STATE";
        field {
          name = "addr1 field";
          desc = "
            HW Status field
          ";
          sw=r; 
        } addr1 [10:0];
        
        field {
          name = "addr0 field";
          desc = "
            HW Status field
          ";
          sw=r; 
        } addr0 [26:16];
        
      } buffer[4] @0x0;
      
    } cp_fp_wr @0x21c0;
    
    // register roll32_counter_reg
    reg {
    
      name = "roll32_counter_reg register";
      field {
        name = "count field";
        sw=rw; 
      } count [3:0] =0x0;
      
    } roll32_counter_reg @0x21d0;
    
    // register sat32_counter_reg
    reg {
    
      name = "sat32_counter_reg register";
      dontcompare;
      field {
        name = "count field";
        dontcompare;
        sw=rw; rclr;
      } count [3:0] =0x0;
      
    } sat32_counter_reg @0x21d4;
    
    // register rcnt_sat_log
    reg {
    
      name = "rcnt_sat_log register";
      field {
        name = "count field";
        sw=rw; rclr;
      } count [15:0] =0x0;
      
      field {
        name = "str field";
        sw=rw; rclr;
        sub_category = "INFO";
      } str [25:16] =0x0;
      
      field {
        name = "subch field";
        sw=rw; rclr;
      } subch [30:26] =0x0;
      
      field {
        name = "en field";
        desc = "
          SW Configuration field
        ";
        sw=rw; 
      } en [31:31] =0x1;
      
    } rcnt_sat_log @0x21d8;
    
  } rdr @0x4000;
  
  // register extra_reg
  reg {
  
    name = "extra_reg name";
    desc = "
      extra_reg description bla bla bla
                more bla
    ";
    category = "DYNAMIC_CONFIG";
    field {
      name = "value field";
      desc = "
        SW Configuration field
      ";
      sw=rw; 
    } value [31:0];
    
  } extra_reg @0x61dc;
  
} simple1;

