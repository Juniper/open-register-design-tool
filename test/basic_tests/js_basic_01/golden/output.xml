<?xml version="1.0" encoding="UTF-8"?>
<!--   Ordt 191120.01 autogenerated file 
       Input: ./js_basic_01/test.js
       Parms: ./js_basic_01/test.parms
       Date: Wed Nov 20 22:19:42 EST 2019
 -->

<map version="191120.01">
  <id>simple1</id>
  <baseaddr>0</baseaddr>
  <shorttext>Registers for simple1</shorttext>
  <regset>
    <id>stats</id>
    <shorttext>stats register_set</shorttext>
    <baseaddr>0x1000</baseaddr>
    <reg>
      <id>disable_check</id>
      <parentpath>stats</parentpath>
      <shorttext>disable_check register</shorttext>
      <baseaddr>0x1000</baseaddr>
      <width>32</width>
      <reps>2</reps>
      <stride>0x4</stride>
      <field>
        <id>l0_b3</id>
        <shorttext>l0_b3 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>6</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>l0_b2</id>
        <shorttext>l0_b2 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>5</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>l0_b1</id>
        <shorttext>l0_b1 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>4</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>l0_b0</id>
        <shorttext>l0_b0 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>3</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>lut_st</id>
        <shorttext>lut_st field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>2</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>cbuf_fifo</id>
        <shorttext>cbuf_fifo field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>1</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>timer_sram</id>
        <shorttext>timer_sram field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>1</width>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>features</id>
      <parentpath>stats</parentpath>
      <shorttext>features register</shorttext>
      <baseaddr>0x1100</baseaddr>
      <width>32</width>
      <field>
        <id>version</id>
        <shorttext>version field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x2</reset>
        <lowidx>30</lowidx>
        <width>2</width>
      </field>
      <field>
        <id>syn_logged</id>
        <shorttext>syn_logged field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>9</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>wdth</id>
        <shorttext>wdth field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x7</reset>
        <lowidx>4</lowidx>
        <width>5</width>
      </field>
      <field>
        <id>generator</id>
        <shorttext>generator field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x1</reset>
        <lowidx>3</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>checker</id>
        <shorttext>checker field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x1</reset>
        <lowidx>2</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>correctable</id>
        <shorttext>correctable field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>1</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>adr_logged</id>
        <shorttext>adr_logged field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x1</reset>
        <lowidx>0</lowidx>
        <width>1</width>
      </field>
      <access>RO</access>
    </reg>
    <reg>
      <id>log_address1</id>
      <parentpath>stats</parentpath>
      <shorttext>log_address1 register</shorttext>
      <baseaddr>0x1104</baseaddr>
      <width>32</width>
      <longtext><![CDATA[
        newdesc bla bla
      ]]></longtext>
      <field>
        <id>value</id>
        <shorttext>value field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>10</width>
        <hwmod></hwmod>
      </field>
      <catcode>32</catcode>
      <access>RW</access>
    </reg>
    <reg>
      <id>log_address2</id>
      <parentpath>stats</parentpath>
      <shorttext>log_address2 register</shorttext>
      <baseaddr>0x1110</baseaddr>
      <width>32</width>
      <longtext><![CDATA[
        newdesc 2 bla bla
      ]]></longtext>
      <field>
        <id>value</id>
        <shorttext>value field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>10</width>
        <hwmod></hwmod>
      </field>
      <catcode>32</catcode>
      <access>RW</access>
    </reg>
    <reg>
      <id>wide_reg</id>
      <parentpath>stats</parentpath>
      <shorttext>wide_reg register</shorttext>
      <baseaddr>0x1120</baseaddr>
      <width>128</width>
      <field>
        <id>value</id>
        <shorttext>value field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0xa5a5a5a5a5a5a5a5a5a5a5a5</reset>
        <lowidx>0</lowidx>
        <width>96</width>
      </field>
      <access>RW</access>
    </reg>
    <highaddr>0x112f</highaddr>
  </regset>
  <regset>
    <id>rdr</id>
    <shorttext>rdr register_set</shorttext>
    <baseaddr>0x4000</baseaddr>
    <reg>
      <id>reorder_window</id>
      <parentpath>rdr</parentpath>
      <shorttext>reorder_window register</shorttext>
      <baseaddr>0x4000</baseaddr>
      <width>32</width>
      <field>
        <id>ws_enable</id>
        <shorttext>ws_enable field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
        </hwinfo>
        <reset>0x1</reset>
        <lowidx>31</lowidx>
        <width>1</width>
        <hwmod></hwmod>
        <longtext><![CDATA[
          SW Configuration field
        ]]></longtext>
      </field>
      <field>
        <id>set_back</id>
        <shorttext>set_back field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
        </hwinfo>
        <reset>0x10</reset>
        <lowidx>16</lowidx>
        <width>11</width>
        <hwmod></hwmod>
        <longtext><![CDATA[
          SW Configuration field
        ]]></longtext>
      </field>
      <field>
        <id>size</id>
        <shorttext>size field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
        </hwinfo>
        <reset>0x80</reset>
        <lowidx>0</lowidx>
        <width>11</width>
        <hwmod></hwmod>
        <longtext><![CDATA[
          SW Configuration field
        ]]></longtext>
      </field>
      <catcode>32</catcode>
      <access>RW</access>
    </reg>
    <reg>
      <id>stream_reconf</id>
      <parentpath>rdr</parentpath>
      <shorttext>stream_reconf register</shorttext>
      <baseaddr>0x4004</baseaddr>
      <width>32</width>
      <field>
        <id>busy</id>
        <shorttext>busy field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>16</lowidx>
        <width>1</width>
        <longtext><![CDATA[
          HW Status field
        ]]></longtext>
      </field>
      <field>
        <id>stream</id>
        <shorttext>stream field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>10</width>
        <longtext><![CDATA[
          SW Configuration field
        ]]></longtext>
      </field>
      <access>RW</access>
    </reg>
    <regset>
      <id>cp_fp_wr</id>
      <parentpath>rdr</parentpath>
      <shorttext>cp_fp_wr register_set</shorttext>
      <baseaddr>0x61c0</baseaddr>
      <reg>
        <id>buffer</id>
        <parentpath>rdr.cp_fp_wr</parentpath>
        <shorttext>buffer register</shorttext>
        <baseaddr>0x61c0</baseaddr>
        <width>32</width>
        <reps>4</reps>
        <stride>0x4</stride>
        <field>
          <id>addr0</id>
          <shorttext>addr0 field</shorttext>
          <access>RO</access>
          <hwinfo>
            <hwaccess>rw</hwaccess>
            <we></we>
          </hwinfo>
          <lowidx>16</lowidx>
          <width>11</width>
          <hwmod></hwmod>
          <longtext><![CDATA[
            HW Status field
          ]]></longtext>
        </field>
        <field>
          <id>addr1</id>
          <shorttext>addr1 field</shorttext>
          <access>RO</access>
          <hwinfo>
            <hwaccess>rw</hwaccess>
            <we></we>
          </hwinfo>
          <lowidx>0</lowidx>
          <width>11</width>
          <hwmod></hwmod>
          <longtext><![CDATA[
            HW Status field
          ]]></longtext>
        </field>
        <catcode>32</catcode>
        <access>RO</access>
      </reg>
      <highaddr>0x61cf</highaddr>
    </regset>
    <reg>
      <id>roll32_counter_reg</id>
      <parentpath>rdr</parentpath>
      <shorttext>roll32_counter_reg register</shorttext>
      <baseaddr>0x61d0</baseaddr>
      <width>32</width>
      <field>
        <id>count</id>
        <shorttext>count field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>4</width>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>sat32_counter_reg</id>
      <parentpath>rdr</parentpath>
      <shorttext>sat32_counter_reg register</shorttext>
      <baseaddr>0x61d4</baseaddr>
      <width>32</width>
      <field>
        <id>count</id>
        <shorttext>count field</shorttext>
        <access>WRC</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>4</width>
        <dontcompare></dontcompare>
      </field>
      <access>WRC</access>
    </reg>
    <reg>
      <id>rcnt_sat_log</id>
      <parentpath>rdr</parentpath>
      <shorttext>rcnt_sat_log register</shorttext>
      <baseaddr>0x61d8</baseaddr>
      <width>32</width>
      <field>
        <id>en</id>
        <shorttext>en field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x1</reset>
        <lowidx>31</lowidx>
        <width>1</width>
        <longtext><![CDATA[
          SW Configuration field
        ]]></longtext>
      </field>
      <field>
        <id>subch</id>
        <shorttext>subch field</shorttext>
        <access>WRC</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>26</lowidx>
        <width>5</width>
      </field>
      <field>
        <id>str</id>
        <shorttext>str field</shorttext>
        <access>WRC</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>16</lowidx>
        <width>10</width>
        <subcatcode>1</subcatcode>
      </field>
      <field>
        <id>count</id>
        <shorttext>count field</shorttext>
        <access>WRC</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>16</width>
      </field>
      <access>RW</access>
    </reg>
    <highaddr>0x61db</highaddr>
  </regset>
  <reg>
    <id>extra_reg</id>
    <shorttext>extra_reg name</shorttext>
    <baseaddr>0x61dc</baseaddr>
    <width>32</width>
    <longtext><![CDATA[
      extra_reg description bla bla bla
                more bla
    ]]></longtext>
    <field>
      <id>value</id>
      <shorttext>value field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <lowidx>0</lowidx>
      <width>32</width>
      <longtext><![CDATA[
        SW Configuration field
      ]]></longtext>
    </field>
    <catcode>2</catcode>
    <access>RW</access>
  </reg>
</map>
